cpp/src/decisiontree/batched-levelalgo/builder_base.cuh

```diff
@@ -323,6 +329,10 @@ struct Builder {
     // start fresh on the number of *new* nodes created in this batch
     CUDA_CHECK(cudaMemsetAsync(n_nodes, 0, sizeof(IdxT), s));
     initSplit<DataT, IdxT, Traits::TPB_DEFAULT>(splits, batchSize, s);
+
+    for (int i = 0; i < batchSize; i++) {
+      h_nodes[node_start + i].info.unique_id = i + node_start;
```

Comment:
Why do we need a new member variable called `unique_id` if it's ID is basically node-id? maybe we could compute this ID inside the kernel itself by passing `node_start` into it and then evaluating it as `blockIdx.z + node_start`? Or am I missing something here?

Reply:
The `unique_id` would be useful for debugging. Say you observe some issue with the way split is happening at node with id 20, you could then enable debug messaging (going in a separate PR) for that particular node. Note that printing happens after training is complete, so it wont be possible to correctly derive node ids at that time.

Reply:
that's a good point. But since we are adding a new member variable to the user-visible `SparseTreeNode` struct, we need to make sure that we still can "import" a RF model that was trained before this change? @JohnZed do you see any issues?

Reply:
We don't make a guarantee of importability across versions of the cuml RF layout. Users can serialize via treelite if they need a persistent version.

I don't have a strong feeling that unique_id is necessary, but if it has been helpful to Vinay in debugging, that would be good evidence to me.